src/test/ui/traits/reservation-impls/reservation-impl-non-lattice-ok.rs

```diff
@@ -0,0 +1,56 @@
+// build-pass
+
+// Check that a reservation impl does not force other impls to follow
+// a lattice discipline.
+
+// Why did we ever want to do this?
+//
+// We want to eventually add a `impl<T> From<!> for T` impl. That impl conflicts
+// with existing impls - at least the `impl<T> From<T> for T` impl. There are
+// 2 ways we thought of for dealing with that conflict:
+//
+// 1. Using specialization and doing some handling for the overlap. The current
+// thought is for something like "lattice specialization", which means providing
+// an (higher-priority) impl for the intersection of every 2 conflicting impls
+// that determines what happens in the intersection case. That's the first
+// thing we thought about - see e.g.
+// https://github.com/rust-lang/rust/issues/57012#issuecomment-452150775
+//
+// 2. The other way is to notice that `impl From<!> for T` is basically a marker
+// trait, as you say since its only method is uninhabited, and allow for "marker
```

Comment:
You should remove 'as you say', since it doesn't make much sense when not in a comment thread :P